lib/node_modules/@stdlib/array/base/linspace2d/lib/main.js

```diff
@@ -0,0 +1,82 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var zeros2d = require( '@stdlib/array/base/zeros2d' );
+
+
+// MAIN //
+
+/**
+* Generates a linearly spaced two-dimensional nested numeric array.
+*
+* @param {number} start - first array value
+* @param {number} stop - last array value
+* @param {NonNegativeIntegerArray} shape - array shape
+* @param {boolean} colexicographic - specifies whether the array values should be in colexicographic order
+* @returns {Array} linearly spaced two-dimensional nested numeric array
+*
+* @example
+* var x = linspace2d( 0, 100, [ 2, 3 ], false );
+* // returns [ [ 0, 20, 40 ], [ 60, 80, 100 ] ]
+
+* x = linspace2d( 0, 100, [ 2, 3 ], true );
+* // returns [ [ 0, 40, 80 ], [ 20, 60, 100 ] ]
+*/
+function linspace2d( start, stop, shape, colexicographic ) {
+	var out;
+	var idx;
+	var s0;
+	var i0;
+	var s1;
+	var i1;
+	var d;
+	var n;
+
+	s0 = shape[ 1 ];
+	s1 = shape[ 0 ];
+
+	n = s0 * s1;
+	if ( n === 0 ) {
+		return [];
+	}
+
+	d = ( stop - start ) / ( n - 1 );
+	out = zeros2d( shape );
```

Comment:
You've effectively implemented this as a two-pass algorithm; however, you can implement this in a single pass. The secret is borrowing ideas from the ndarray kernels. Namely, based on the value of the colexicographic argument, you set two increment variables: one for the inner loop and one for the outer loop. The inner loop controls how values are incremented along a row. The outer loop variable controls how to increment to the first value along that row.

Using this approach, you can also remove the inner `colexicographic` branch.